### PR TITLE
Documented HTTP-Methods and added HEAD-Method

### PR DESCRIPTION
**Pull request by Monarchis**
Added and documented some more HTTP-Methods

Just to update the existing ones.

### CODE REVIEW

**Comment by fabiitch** [comment]
nice !
just need to edit changelog i think.

**Comment by intrigus** [comment]
Do these work on gwt at all?

**Comment by Monarchis** [comment]
@intrigus sur ethis is just adding new fields to a static class that had not all HTTP-Methods as constants before, (all checks passed)

**Comment by noblemaster** [comment]
There is more to it than just adding a few constants. Please look at the code:
https://github.com/libgdx/libgdx/blob/master/gdx/src/com/badlogic/gdx/net/NetJavaImpl.java#L175

**Comment by SimonIT** [comment]
To make it work on gwt, there has to be made some changes for the new methods, if they are supported. Else, it should be documented that they are not supported on gwt
https://github.com/libgdx/libgdx/blob/11a53e2d9d573e2a0a94457b309dbf7044623497/backends/gdx-backends-gwt/src/com/badlogic/gdx/backends/gwt/GwtNet.java#L129-L145

**Comment by MobiDevelop** [comment]
On GWT, we use RequestBuilder, which supports `DELETE`, `GET`, `HEAD`, `POST`, `PUT`. Any others should be documented as unavailable on GWT.

**Comment by noblemaster** [comment]
Unless GWT can support the other methods (sorry not too familiar?), it probably makes sense to drop this pull request for compatiblity's sake?

**Comment by Monarchis** [comment]
@SimonIT
I just extended the constants, but you are free to implement it.
This Line should tell that it's not supported:
https://github.com/libgdx/libgdx/blob/11a53e2d9d573e2a0a94457b309dbf7044623497/backends/gdx-backends-gwt/src/com/badlogic/gdx/backends/gwt/GwtNet.java#L144

**Comment by Monarchis** [comment]
@noblemaster Compatibility does not break

File: gdx/src/com/badlogic/gdx/Net.java [hunk 1]
```diff
@@ -93,20 +93,92 @@
...
+		 * The CONNECT method establishes a tunnel to the server
+		 * identified by the target resource.
+		 * Not Supported by libgdx yet.
+		 **/
+		public static final String CONNECT = "CONNECT";
+
+		/**
+		 * The TRACE method performs a message loop-back test along
+		 * the path to the target resource.
+		 * Not Supported by libgdx yet.
```
Comment (MobiDevelop) [inline gdx/src/com/badlogic/gdx/Net.java]: Adding things only to say they aren't supported isn't super nice.

Comment (Monarchis) [reply gdx/src/com/badlogic/gdx/Net.java]: @MobiDevelop well, then reject this pr

Comment (noblemaster) [reply gdx/src/com/badlogic/gdx/Net.java]: I agree. I think better would be to update Net.java to accept any method string also allowing to specify if output is desired or not. After all, GET or POST could both be used with our without stream output, even though not commonly used for GET as opposed to POST.

Comment (Monarchis) [reply gdx/src/com/badlogic/gdx/Net.java]: @noblemaster I'm abled to add some of that methods easy, but connection and trace need more efford.

Comment (Monarchis) [reply gdx/src/com/badlogic/gdx/Net.java]: Getting Head to work with GWT, but the others are not supported right now, I'll mark it.

**Comment by noblemaster** [comment]
@Monarchis If you want this pull request to continue, please (1) make sure those methods work on all backends. Also, (2) given we add a lot of methods, let's also allow custom HTTP methods in addition to the ones you proposed supporting both ones that do output (ditto POST) and those that don't (ditto GET). I think that make the most sense.

Also, don't forget to (3) write JUnit test for your methods and the custom methods testing them on all backends.

**Comment by Monarchis** [comment]
@noblemaster The NetGwt is using google lib that does not support PATCH; OPTIONS, TRACE and CONNECT so there will be a problem mapping them to the other 5 existing ones, HEAD is possible to implement properly.

**Comment by Monarchis** [comment]
One thing that i can do is marking the one that do not work on GWT as not supported by gtw, and removing the rest that does not work right now.

**Comment by noblemaster** [comment]
I see. In that sense it probably makes little sense to add them besides probably HEAD which is of little use if you ask me. libGDX shouldn't offer methods that are not supported on all backends.

**Comment by Monarchis** [comment]
@noblemaster 
They can be implemented on all Platforms, only gwt has problems.

**Comment by Monarchis** [comment]
ok only HEAD is left as change.

**Comment by noblemaster** [comment]
There is some duplicate code you added in `GwtNet`. Also, what about the unit tests for all the platform to make sure it actually works?

**Comment by Monarchis** [comment]
@noblemaster the only thing i added was the HEAD-Method, and setting the UriConnection.setDoInput() to false if it's head. (GwtNet).
The redundancies, idk maybe i'll remove them.

But currently I'm still not sure if head also needs an Input (the headers that are responded to the client).

May i really have to test this Method, otherwise, nothing else has changed.

**Review by MrStahlfelge — APPROVED** [review]
Last change is a good change. It is non-breaking, an enhancement and should be sqash merged.

**Pull request merged**